src/main/java/dan200/computercraft/core/apis/TermAPI.java

```diff
@@ -101,6 +111,59 @@ public static int parseColour( Object[] args, boolean _enableColours ) throws Lu
         };
     }
 
+    public static void setColour( Terminal terminal, int colour, float r, float g, float b )
+    {
+        if( terminal.getPalette() != null )
+        {
+            terminal.getPalette().setColour( colour, r, g, b );
+            terminal.setChanged();
+        }
+    }
+
+    public static void setColour( Terminal terminal, HashMap<Object, Object> colours) throws LuaException
```

Comment:
I'm not convinced we need/want the ability to pass in a table. Calling the method 16 times for a complete palette swap should be fine.